pkg/library/testdata/lifecycle/prestart_exec_command.yaml

```diff
@@ -0,0 +1,29 @@
+name: "Should return init container with prestart exec command"
+
+input:
+  components:
+    - container:
+        name: test-container1
+        image: my-image
+    - container:
+        name: test-container2
+        image: my-image
+  commands:
+    - exec:
+        id: test_command
+        component: test-container1
+        command: "test_command"
+  events:
+    preStart:
+      - "test_command"
+
+output:
+  initContainers:
```

Comment:
Should not be it ignored if `errRegexp` is specified?

Reply:
Adding such condition in test code could help us keep test data clean, but that's not simple, so up to you to add it or not
```go
//both of error and containers should not be configured at the same time
if tt.Output.ErrRegexp != nil && len(*tt.Output.ErrRegexp) > 0 &&
    (len(tt.Output.InitContainers) > 0 || len(tt.Output.MainContainers) > 0) {
    t.Fatal("No init containers or main containers could be configured when error is expected")
}
```

Reply:
In general the convention is you can't make any assumptions about non-error return value when `err != nil`. The leftovers in this test case are due to disabling preStart exec support after having written the tests. I've removed the initContainers/mainContainers section for this file.